frontend/appflowy_tauri/src/appflowy_app/components/document/TextActionMenu/menu/BgColorPicker.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import ColorPicker from '$app/components/document/TextActionMenu/menu/ColorPicker';
+import { FormatColorFill, FormatColorText } from '@mui/icons-material';
+import { TextAction } from '$app/interfaces/document';
+
+function BgColorPicker() {
+  const { t } = useTranslation();
+
+  const getColorIcon = useCallback((color: string) => {
+    return (
+      <div
+        style={{
+          backgroundColor: color,
+        }}
+        className={'rounded border border-line-divider p-0.5'}
+      >
+        <FormatColorText />
+      </div>
+    );
+  }, []);
+  const colors = useMemo(
+    () => [
+      {
+        name: t('colors.default'),
+        key: 'default',
+        color: 'transparent',
+      },
+      {
+        name: t('colors.custom'),
+        key: 'custom',
+        color: 'transparent',
+      },
+      {
+        key: 'gray',
+        name: t('colors.gray'),
+        color: '#78909c',
+      },
```

Comment:
Currently, all of these colors exist multiple times.

I think it would be nice to have them defined somewhere such that any type of color picker can use them.

Reply:
> Currently, all of these colors exist multiple times.

I don't quite understand, where exactly?

Reply:
Not really sure where the best location would be since I'm not too into the architecture and abstraction of the tauri app.

But something like this:
```
import { TFunction } from "react-i18next";

function getColorOptions(TFunction t)  {
  return [
    {
      name: t('colors.default'),
      key: 'default',
      color: 'transparent',
    },
    ...
  ];
}
```

Reply:
In fact, this entry file is used for configuration, and the real component is ColorPicker
After using useMemo, if `t` has not changed, this object is cached

Reply:
Ah okay, it's just that they're the same, I was expecting the color options would be 1:1 even in the future. (Eg. if a user adds a custom color to their palette, it should be possible to pick that for both background and text.).

Reply:
The background color and text color have different values now, so I separate their entry files